app/services/ConsignmentStatusService.scala

```diff
@@ -13,13 +15,37 @@ import scala.concurrent.{ExecutionContext, Future}
 class ConsignmentStatusService @Inject() (val graphqlConfiguration: GraphQLConfiguration)(implicit val ec: ExecutionContext) {
   private val getConsignmentStatusClient = graphqlConfiguration.getClient[gcs.Data, gcs.Variables]()
 
-  def getConsignmentStatus(consignmentId: UUID, token: BearerAccessToken): Future[Option[GetConsignment.CurrentStatus]] = {
+  def getStatusValues(statuses: List[ConsignmentStatuses], statusTypes: StatusType*): Map[StatusType, Option[String]] = {
+    statusTypes
+      .map(t => {
+        val value = statuses.find(_.statusType == t.id) match {
```

Comment:
This is personal preference so feel free to ignore but you don't need the case statement, you can just to `statuses.find(_.statusType == t.id).map(_.value)`